cmd/pinsharesvc/service.go

```diff
@@ -0,0 +1,280 @@
+				s.logInfo("Service stop requested")
+				statusChan <- svc.Status{State: svc.StopPending}
+				s.shutdown()
+				break loop
+			case svc.Pause:
+				statusChan <- svc.Status{State: svc.Paused, Accepts: cmdsAccepted}
+				s.logInfo("Service paused")
+			case svc.Continue:
+				statusChan <- svc.Status{State: svc.Running, Accepts: cmdsAccepted}
+				s.logInfo("Service continued")
+			default:
+				s.logError(fmt.Sprintf("Unexpected control request #%d", c), nil)
+			}
+		case <-s.ctx.Done():
+			s.logInfo("Service context cancelled")
+			break loop
+		}
+	}
+
+	// Wait for shutdown to complete
+	s.wg.Wait()
+	statusChan <- svc.Status{State: svc.Stopped}
+	return false, 0
+}
+
+// initialize sets up all service components
+func (s *pinshareService) initialize() error {
+	s.ctx, s.cancel = context.WithCancel(context.Background())
+
+	// Load configuration
+	config, err := LoadConfig()
+	if err != nil {
+		return fmt.Errorf("failed to load configuration: %w", err)
+	}
+	s.config = config
+
+	// Initialize event log
+	s.eventLog, err = openEventLog(serviceName)
+	if err != nil {
+		return fmt.Errorf("failed to open event log: %w", err)
+	}
+
+	s.logInfo(fmt.Sprintf("Loaded configuration: DataDir=%s, IPFSPort=%d, APIPort=%d",
+		config.DataDirectory, config.IPFSAPIPort, config.PinShareAPIPort))
+
+	// Ensure directories exist
+	if err := s.config.EnsureDirectories(); err != nil {
+		return fmt.Errorf("failed to create directories: %w", err)
+	}
+
+	// Initialize process manager
+	s.processManager = NewProcessManager(s.config, s.eventLog)
+
+	// Initialize health checker before starting processes (needed for health checks during startup)
+	s.logInfo("Initializing health checker...")
+	s.healthChecker = NewHealthChecker(s.config, s.processManager, s.eventLog)
+
+	// Start IPFS daemon
+	s.logInfo("Starting IPFS daemon...")
+	if err := s.processManager.StartIPFS(s.ctx); err != nil {
+		return fmt.Errorf("failed to start IPFS: %w", err)
+	}
+
+	// Wait for IPFS to be ready
+	s.logInfo("Waiting for IPFS to become ready...")
+	if err := s.waitForIPFS(); err != nil {
+		return fmt.Errorf("IPFS failed to start: %w", err)
+	}
+	s.logInfo("IPFS daemon is ready")
+
+	// Start PinShare backend
+	s.logInfo("Starting PinShare backend...")
+	if err := s.processManager.StartPinShare(s.ctx); err != nil {
+		return fmt.Errorf("failed to start PinShare: %w", err)
+	}
+
+	// Wait for PinShare to be ready
+	s.logInfo("Waiting for PinShare API to become ready...")
+	if err := s.waitForPinShare(); err != nil {
+		return fmt.Errorf("PinShare failed to start: %w", err)
+	}
+	s.logInfo("PinShare backend is ready")
+
+	// Start embedded UI server
+	s.logInfo("Starting embedded UI server...")
+	s.uiServer = NewUIServer(s.config, s.eventLog)
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+		if err := s.uiServer.Start(s.ctx); err != nil {
+			s.logError("UI server error", err)
+		}
+	}()
+	s.logInfo(fmt.Sprintf("UI server started on http://localhost:%d", s.config.UIPort))
```

Comment:
Let's comment this out like we did for the other UI related code that is waiting for the UI to be merged.